Info Display enhancement : Rotation, Rotation Speed, TimeDifference, Oriented Speed.

Blounard opened the pull request:
Oriented Speed is the XZ speed, but instead of decomposing it as X and Z, it's decomposed as Forward and Sideway (relative to your facing angle)
This PR also contain the advanced ghost comp, but the implementation might change in the future.

Blounard committed:
Rotation

1st attempt at adding rotation to infodisplay. Unfinished. Mainly a test of using github

Blounard committed:
Rotation + Rotation speed

Added rotation and rotation speed to the infodisplay. Should be ready to use.

Blounard committed:
Bugfix

Fixed an issue in race manager : RaceState is 4 when the race is finished, not 3. (atleast for TTs). It was previously causing the info display to turn off on the race finish state, despite the code not aiming to do that.

Fixed the Rotation speed calculation to not trigger error while not in a race. (It wasn't a big issue, it was just spamming the python log)

Blounard committed:
info_display timediff

Added time difference to info display
Changed some code for the rotation speed.
Added a class to store data from previous frames.
Added Oriented speed to infodisplay

Blounard committed:
TimeDiff Update + AGC + Bruteforcer

Small changes to TimeDiff, to make the code more clean, and do less computations.

Added Advanced Ghost Comp. (might still change a bit)

Added a Bruteforcer (still wip)

Blounard committed:
Added doc and removed personal files

malleoz committed:
Fix staticmethod for outside_drift_last_dir (#56)

ximk commented on .gitignore:
```diff
@@ -1 +1,6 @@
 /scripts/MKW_Inputs/*.csv
+scripts/Modules/test_del_me_later.py
+scripts/Modules/test.bit
+scripts/Modules/test_del_me_later.py
+scripts/Modules/test_del_me_later.py
```
Should be deleted prior to merge.

ximk commented on scripts/Modules/infodisplay.ini:
```diff
@@ -0,0 +1,39 @@
+[DEBUG]
+debug = True
```
Entire file should be taken out of this PR prior to merge to avoid conflict.

ximk commented on scripts/Modules/mkw_utils.py:
```diff
@@ -81,5 +122,215 @@ def get_unrounded_time(lap, player):
+    return speed_to_euler_angle(speed)
+
+
+# TODO: Time difference display helper functions
+"""The time difference functions.
+time_difference_[name](P1, S1, P2, S2) is a function that takes as arguments
+P1,S1 : Player1's Position and Speed vec3.
+P2,S2 : Player2's Position and Speed vec3
+Return the time it would take for Player1 to catch Player2 (not always symmetric)
+
+get_time_difference_[name](Player1, Player2) takes as arguments
+Player1 : Player1 ID
+Player2 : Player2 ID
+Return the time it would take for Player1 to catch Player2 (not always symmetric)
+It's the function called in draw_infodisplay.py
+"""
```
Is this WIP?

ximk reviewed (changes requested):
Writing a pre-test review to make sure these comments are addressed prior to merge.

ximk committed:
Remove compiled code from PR.

Dolphin will do this already, no need to clutter the PR with this.

ximk commented on scripts/Modules/agc_lib.py:
```diff
@@ -0,0 +1,297 @@
+        
+def file_to_framedatalist(filename):
+    datalist = []
+    file = open(filename, 'r')
+    if file is None :
+        gui.add_osd_message("Error : could not load the data file")
+    else:
+        timerdata = None
+        metadata = None
+        rkg_metadata = None
+        listlines = file.readlines()
+        if listlines[0][0] == 'r':           
+            rkg_metadata = FrameData(string = listlines[0][1:])
+            timerdata = TimerData.from_sliced_rkg(rkg_metadata)
+        else:
+            metadata = FrameData(string = listlines[0])
+            if listlines[-1][0]=='t':
+                timerdata = TimerData(string = listlines.pop()[1:])
+        for i in range(1, len(listlines)):
+            datalist.append(FrameData(string = listlines[i]))
+        file.close()
+        gui.add_osd_message(f"Data successfully loaded from {filename}")
+        return metadata, datalist, timerdata, rkg_metadata
+
+
+def framedatalist_to_file(filename, datalist, rid):
+    metadata = get_metadata(rid)
+    timerdata = get_timerdata(rid)
+    file = open(filename, 'w')
+    if file is None :
+        gui.add_osd_message("Error : could not create the data file")
+    else:
+        file.write(str(metadata))
+        for frame in range(max(datalist.keys())+1):
+            if frame in datalist.keys():
+                file.write(str(datalist[frame]))
+            else:
+                file.write(str(FrameData(get_addr(rid), usedefault=True)))
+        file.write(str(timerdata))
+        file.close()
```
I think sticking to just `read_id` or `rid` would be best here, unless this is necessary for it to work.

ximk replied:
> I think sticking to just `read_id` or `rid` would be best here, unless this is necessary for it to work.

```suggestion
def timerdata_to_file(filename, read_id):
    timerdata = TimerData(read_id)
    file = open(filename, 'a')
    if file is None :
        gui.add_osd_message("Error : could not create the data file")
    else :
        file.write(str(timerdata))
        file.close()

def get_timerdata(read_id):
    return TimerData(read_id)
        
def file_to_framedatalist(filename):
    datalist = []
    file = open(filename, 'r')
    if file is None :
        gui.add_osd_message("Error : could not load the data file")
    else:
        timerdata = None
        metadata = None
        rkg_metadata = None
        listlines = file.readlines()
        if listlines[0][0] == 'r':           
            rkg_metadata = FrameData(string = listlines[0][1:])
            timerdata = TimerData.from_sliced_rkg(rkg_metadata)
        else:
            metadata = FrameData(string = listlines[0])
            if listlines[-1][0]=='t':
                timerdata = TimerData(string = listlines.pop()[1:])
        for i in range(1, len(listlines)):
            datalist.append(FrameData(string = listlines[i]))
        file.close()
        gui.add_osd_message(f"Data successfully loaded from {filename}")
        return metadata, datalist, timerdata, rkg_metadata


def framedatalist_to_file(filename, datalist, read_id):
    metadata = get_metadata(read_id)
    timerdata = get_timerdata(read_id)
    file = open(filename, 'w')
    if file is None :
        gui.add_osd_message("Error : could not create the data file")
    else:
        file.write(str(metadata))
        for frame in range(max(datalist.keys())+1):
            if frame in datalist.keys():
                file.write(str(datalist[frame]))
            else:
                file.write(str(FrameData(get_addr(rid), usedefault=True)))
        file.write(str(timerdata))
        file.close()
```

ximk commented on scripts/Modules/mkw_utils.py:
```diff
@@ -81,5 +122,215 @@ def get_unrounded_time(lap, player):
+    curframe = history.get_older_frame(0)
+    lastframe = history.get_older_frame(-1)
+    inf = float('inf')
+    if curframe.prc>=curframe.grc:
+        if curframe.grc>lastframe.prc:
+            l = [history.get_older_frame(k).prc for k in range(history.size)]
+            i = find_index(curframe.grc, l)
+            t = i + (curframe.grc - l[i])/ (l[i+1] - l[i])
+            return -t
+        return -inf
+    else:
+        if curframe.prc>lastframe.grc:
+            l =[history.get_older_frame(k).grc for k in range(history.size)]
+            i = find_index(curframe.prc, l)
+            t = i + (curframe.prc - l[i])/ (l[i+1] - l[i])
+            return t
+        return inf
```
I believe the linter does not like some of the variable names, however this should fix it.
```suggestion
    if curframe.prc>=curframe.grc:
        if curframe.grc>lastframe.prc:
            length = [history.get_older_frame(k).prc for k in range(history.size)]
            i = find_index(curframe.grc, l)
            time = i + (curframe.grc - length[i])/ (length[i+1] - length[i])
            return -time
        return -inf
    else:
        if curframe.prc>lastframe.grc:
            length = [history.get_older_frame(k).grc for k in range(history.size)]
            i = find_index(curframe.prc, l)
            time = i + (curframe.prc - length[i])/ (length[i+1] - length[i])
            return time
        return inf
```

ximk commented on scripts/RMC/Adv. Ghost Comp/Load_data.py:
```diff
@@ -0,0 +1,54 @@
+from dolphin import event, gui, utils
```
The GUI module is not used in this file.
```suggestion
from dolphin import event, utils
```

ximk commented on scripts/RMC/Adv. Ghost Comp/Load_data.py:
```diff
@@ -0,0 +1,54 @@
+
+    metadata, framedatalist, timerdata, rkg_metadata = lib.file_to_framedatalist(filename)
+
+    if not timerdata is None:
```
```suggestion
    if timerdata is not None:
```

ximk commented on scripts/RMC/Adv. Ghost Comp/Load_data.py:
```diff
@@ -0,0 +1,54 @@
+    delayed_frame = floor(delay)+frame
+    decimal_delay = delay - floor(delay)
+
+    if not metadata is None:
```
```suggestion
    if metadata is not None:
```

ximk commented on scripts/RMC/Adv. Ghost Comp/Load_data.py:
```diff
@@ -0,0 +1,54 @@
+        metadata.write(lib.get_metadata_addr(1))
+        
+    if lib.is_rkg():   
+        if not rkg_metadata is None:
```
```suggestion
        if rkg_metadata is not None:
```

ximk commented on scripts/RMC/Adv. Ghost Comp/Save_to_file.py:
```diff
@@ -0,0 +1,46 @@
+from dolphin import event, gui, utils
```
```suggestion
from dolphin import event, utils
```

ximk commented on scripts/RMC/_draw_info_display.py:
```diff
@@ -3,10 +3,11 @@
 import os
 
-from Modules.mkw_classes.common import SurfaceProperties
+from Modules.mkw_classes.common import SurfaceProperties, eulerAngle
```
```suggestion
from Modules.mkw_classes.common import SurfaceProperties
```

ximk commented on scripts/RMC/_draw_info_display.py:
```diff
@@ -3,10 +3,11 @@
 
 import Modules.mkw_utils as mkw_utils
-from Modules.mkw_classes import RaceManager, RaceManagerPlayer, RaceState
+from Modules.mkw_classes import RaceManager, RaceManagerPlayer, RaceState, TimerManager
```
```suggestion
from Modules.mkw_classes import RaceManager, RaceManagerPlayer, RaceState
```

ximk commented on scripts/Modules/agc_lib.py:
```diff
@@ -0,0 +1,297 @@
+from dolphin import gui, memory
+from .mkw_classes import quatf, vec3
```
```suggestion
from .mkw_classes import vec3
```

ximk commented on scripts/Modules/agc_lib.py:
```diff
@@ -0,0 +1,297 @@
+from dolphin import gui, memory
+from .mkw_classes import quatf, vec3
+from .mkw_classes import VehiclePhysics, KartMove, RaceConfig, Timer, RaceManagerPlayer
```
```suggestion
from .mkw_classes import VehiclePhysics, KartMove, RaceConfig, RaceManagerPlayer
```

ximk reviewed (changes requested):
These changes should appease the linter.

ximk committed:
Appease linter

ximk committed:
Appease linter

ximk committed:
Update scripts/RMC/Adv. Ghost Comp/Load_data.py

ximk committed:
github is being silly

ximk committed:
Final touches

vabold commented on scripts/AGC_Data/save your AGC data here.txt:
If the purpose of this is to have a placeholder file to keep the folder, `.gitkeep` is the preferred name. Otherwise, messages or notes are best done in `.txt` or `.md` file bodies.

Blounard replied:
That's the only purpose of that text file, delete it if it's not needed

vabold commented on .gitignore:
```diff
@@ -1 +1,2 @@
 /scripts/MKW_Inputs/*.csv
+scripts/AGC_Data/ghost.data
```
Consistency.

```suggestion
/scripts/AGC_Data/ghost.data
```

vabold commented on scripts/Modules/TimeDifference information.txt:
Avoid spaces in filenames. `TimeDifferenceInfo.txt` would be my pick.

Optionally, this can be improved via markdown.

vabold commented on scripts/Modules/agc_lib.py:
```diff
@@ -0,0 +1,320 @@
+            memory.write_bytes(addr, val)
+
+
+def float_to_str(f):
+    ms = round((f % 1)*1000)
+    s = math.floor(f) % 60
+    m = math.floor(f)//60
+    return f"{m},{s},{ms}"
+
+
+def floats_to_str(fs):
+    return f"{float_to_str(fs[0])};{float_to_str(fs[1])};{float_to_str(fs[2])}"
```
Context?

Blounard replied:
Old functions that i forgot to remove when I changed the implementation of TimerData. You can delete

vabold commented on scripts/Modules/agc_lib.py:
```diff
@@ -0,0 +1,320 @@
+def get_addr(player_id):
+    a = VehiclePhysics.chain(player_id)
+    b = KartMove.chain(player_id)
+    return [(a+0x68, 12),  # Position
+            (a+0xF0, 16),  # Rotation
+            (a+0x74, 12),  # EV
+            (a+0x14C, 12),  # IV
+            (b+0x18, 4),  # MaxEngineSpd
+            (b+0x20, 4),  # EngineSpd
+            (b+0x9C, 4),  # OutsideDriftAngle
+            (b+0x5C, 12)]  # Dir
```
The problem is that these are blindly referenced (e.g. `addr[0]`, `addr[1]`, etc). There are two alternatives to solve this:

- Return a dict with key-value pairs, where the keys are these comments
- Create a dataclass with a static `read` method

Blounard replied:
I didn't know about static read method was a thing when i wrote that, that's why some init functions are a mess.
This class will probably be rewritten and merged with mkw_utils.FrameData at some point

vabold commented on scripts/Modules/agc_lib.py:
```diff
@@ -0,0 +1,320 @@
+def is_rkg():
+    s = bytearray('RKGD', 'ASCII')
+    r = rkg_addr()
+    return s == memory.read_bytes(r, 4)
```
At this point, it may as well be beneficial to create an RKG wrapper class.

vabold commented on scripts/Modules/mkw_classes/common.py:
```diff
@@ -38,17 +38,55 @@ def __add__(self, other):
     def __sub__(self, other):
         return vec3(self.x - other.x, self.y - other.y, self.z - other.z)
 
+    def __mul__(self, other):
+        """ vec3 * vec3 -> float (dot product)
+            vec3 * float -> vec3 (scalar multiplication)"""
+        if type(other) == vec3:
+            return self.x * other.x + self.y * other.y + self.z * other.z
+        else:
+            return vec3(self.x * other, self.y * other, self.z * other)
```
I'm not a fan of the ambiguous type return. It would likely be better to have a designated `dot` method.

```py
if type(other) == vec3:
    return vec3(self.x * other.x, self.y * other.y, self.z * other.z)
else:
    return vec3(self.x * other, self.y * other, self.z * other)

Blounard replied:
There are 3 types of vector multiplications useful in different context.
The dot product, which takes two vec3 and return a number (float typically)
The cross product, which takes two vec3 and return a vec3
The scalar multiplication, which takes a vec3 and a number, and return a vec3
But I knew only 2 multiplicatives token for python class (being "*" and "@")
So I used "*" for the dot product and the scalar multiplication, and "@" for the cross product and scalar multiplication.
If the ambiguous return type is really a problem, i'd rather have "*" for dot product, "@" for cross product, and a "scalarmult" method for the scalar multiplication.
But I don't mind this ambiguity personally, as it allow for some code very simple to read imo.

vabold commented on scripts/Modules/mkw_classes/common.py:
```diff
@@ -38,17 +38,55 @@ def __add__(self, other):
+        else:
+            return vec3(self.x * other, self.y * other, self.z * other)
+
+    def __matmul__(self, other):
+        """ vec3 @ vec3 -> vec3 (cross product)
+            vec3 @ float -> vec3 (scalar multiplication)"""        
+        if type(other) == vec3:
+            x = self.y*other.z - self.z*other.y
+            y = self.z*other.x - self.x*other.z
+            z = self.x*other.y - self.y*other.x
+            return vec3(x,y,z)
+        else:
+            return vec3(self.x * other, self.y * other, self.z * other)
```
Similarly, a designated `cross` method would also likely be better.

Out of curiosity, what is `__matmul__`?

Blounard replied:
The type ambiguity is less of a problem here since matmul always return a vec3 in this context

__matmul__ is a class method, just like __add__ or __mul__. It uses the token "@".
So you write a@b and it's interpreted as a.__matmul__(b)
Here it design the cross product

vabold commented on scripts/Modules/mkw_classes/common.py:
```diff
@@ -38,17 +38,55 @@ def __add__(self, other):
         bytes = memory.read_bytes(ptr, 0xC)
         return vec3(*struct.unpack('>' + 'f'*3, bytes))
 
+    @staticmethod
+    def from_bytes(bts) -> "vec3":
+        return vec3(*struct.unpack('>' + 'f'*3, bts))
+
+    def to_bytes(self) -> bytearray:
+        return bytearray(struct.pack('>fff', self.x, self.y, self.z))
```
Optional:

```suggestion
    @staticmethod
    def unpack(bts) -> "vec3":
        return vec3(*struct.unpack('>' + 'f'*3, bts))

    def pack(self) -> bytearray:
        return bytearray(struct.pack('>' + 'f'*3, self.x, self.y, self.z))
```

vabold commented on scripts/Modules/mkw_utils.py:
```diff
@@ -1,14 +1,53 @@
 #  these commented-out lines:
 
+
+class FrameData:
```
Could this potentially conflict with `agc_lib.py`?

Blounard replied:
It can conflict if you import both class with the same name.
Currently it's not an issue, but if soemone makes some script using both agc_lib and mkw_utils, it could be.
I'll eventually merge both class into the same class (currently, agc_lib.FrameData store values you can read in the memory, and mkw_utils.FrameData is more messy, but can store function from values you can read in the memory, and only used for the draw_info_display.)

vabold commented on scripts/RMC/Adv. Ghost Comp/Load_data.py:
Please rename the folder to exclude non-alphanumeric characters (e.g. `AdvGhostComp`).

Blounard committed:
Implemented the 7 column version of TTK

 First attempt at implementing TTK with 7 columns

Blounard committed:
Merge branch 'main' of https://github.com/Blounard/mkw-scripts

Blounard committed:
remade agc

remade agc to match lua core version
need to do the load ghost combo stuff still

Blounard committed:
agc other functions

added other agc scripts (instant replay and copy inputs)
made config file for agc
config file for infodisplay and agc now reload on loadstate

Blounard committed:
rkg lib

Added library function for rkg manipulation
Added a RKG auto saver
Added TTK Activate Ghost Hard

Epik95mkw committed:
Remove pycache files

Epik95mkw committed:
Add dolphin function stubs

Epik95mkw committed:
Add glitchy corner accessor

Epik95mkw committed:
Personal infodisplay

Epik95mkw committed:
External scripts and memory sharing utils

Epik95mkw committed:
WIP superhopping macro

Epik95mkw committed:
Superslide (and neutral glide) macro

Epik95mkw committed:
Auto MT release macro

Epik95mkw committed:
WIP teleport offset script

Blounard committed:
Merge pull request #1 from Epik95mkw/main

Epik95mkw/mkw-scripts

Blounard committed:
RKG Loader, TTK RKG to Player, TTK Player to CSV

added some function with an open/save dialog box

Blounard committed:
mkds minimap + other small utils

Made a script to display player and ghost position on a external screen, similar to mkds minimap style

Made some modif to the History class in mkw_utils to make it more polyvalent

Added some utils function for later stuff (teleport related stuff)

Blounard committed:
auto teleport + various bugfix

Blounard committed:
Update .gitignore

Blounard committed:
auto tp componed component

Epik95mkw committed:
Soft drift assist macro

Epik95mkw committed:
Merge remote-tracking branch 'upstream/main'

Epik95mkw committed:
Reorganize stuff

Blounard committed:
update mkds minimap

it now display checkpoints, and have some keyboard options

Epik95mkw committed:
Merge remote-tracking branch 'upstream/main'

Blounard committed:
Big script commit

Added TAS input window
Added TTK Save GUI
Added a batch file to install external python package
Updated start slides. Start slides now just need the left slide.csv to work. (right slide use a mirror of left slide)
Added some stuff to infodisplay
Infodisplay create function are now coded in a lib in Modules, so other scripts can use it too (like external info display)
Added External info display
Fixed a bug with RKG recreation in rkg_lib
Added a freefly script
Fixed a TTK bug where TTK Activate wouldn't always reload the correct .csv on load state

Blounard committed:
Update TAS_Input.py

fixed TAS_Input that was doing nothing.
TAS_Input still has a 1f delay. i'm not sure how to fix it. I think you'd have to keep the ttk_lib run input line always active, instead of being called on frame advance event, but I wasn't able to do it

Epik95mkw committed:
Simple TTK GUI

Epik95mkw committed:
Merge remote-tracking branch 'upstream/main'

Epik95mkw committed:
TTK GUI improvements

Epik95mkw committed:
Small fix

Blounard committed:
Remove all on state load events

Removed all on state load events to avoid crashes on load state.

This commit should be undone (manually if possible to make sure everything works correclty) when the onstateload event will be synchronous with the emu thread.

For now, the expected on state load functions are called when the race frame counter isn't incrementing (or staying unchanged) on frame advance

Blounard committed:
gitignore modif to include default mii

Blounard committed:
Merge commit '5e237f4c04f522274cf6109cd6dc53a2f2bab2b4'

Epik95mkw committed:
Merge remote-tracking branch 'upstream/main'

Epik95mkw committed:
Update alt infodisplay with docs

Epik95mkw committed:
TTK GUI listener thread for button commands

Epik95mkw committed:
Fix TTK GUI crashing on exit

Epik95mkw committed:
Use window.mainloop for gui window

Epik95mkw committed:
Fix freeze(s) caused by TTK GUI buttons

Epik95mkw committed:
Add more comments

Epik95mkw committed:
Update macros and add usage instructions

Epik95mkw committed:
Update wheelie turn macro

Epik95mkw committed:
Update startslide scripts and inputs

Blounard committed:
Merge pull request #2 from Epik95mkw/main

TTK GUI and updated macros

Blounard committed:
Delete scripts/Settings/Infodisplay.ini

Blounard committed:
Delete scripts/Settings/AGC.ini

Blounard committed:
Delete scripts/AGC_Data/ghost_data.agc

Blounard committed:
Delete scripts/MKW_Inputs/Startslides directory

Blounard committed:
Delete scripts/Ghost/Default directory

Blounard committed:
Removed unwanted file from the repo

Deleted and re uploaded some folders, to remove unecessary files.

Blounard committed:
Merge branch 'main' of https://github.com/Blounard/mkw-scripts

Blounard committed:
TTK Save scripts auto cancel

Changed TTK Scripts to auto cancel after saving

Blounard committed:
Improved TTK Activate scripts

On TTK Activate scripts, input are now performed on frame begin instead of frame advance.
Added back the event on state load

Blounard committed:
infodisplay update

Removed april fool prank
updated the infodisplay to work with the 0f delay option
turned back setting reload on stateload
added Glitchy corner float to misc
fixed rounding errors

Blounard committed:
Update TAS_input_window.py

tas input look a tiny bit better

Blounard committed:
Update .gitignore

Blounard committed:
Delete scripts/RMC/hop test.py

Blounard committed:
Delete scripts/RMC/test.py

Blounard committed:
Delete scripts/RMC/set_position_test.py

Blounard committed:
TTK Settings and Backup

Implemented TTK Settings and backup

Blounard committed:
Update AGC_Save_Ghost.py

save on exit
exit on finish race

Blounard committed:
Update draw_input_display.py

updated draw moment to work with 0f UI delay option

Blounard committed:
Update TAS_Input.py

removed 1f delay

Blounard committed:
draw_infodisplay more often on screen

draw infodisplay is now drawn when in a race, instead of when inputs are available

Blounard committed:
TTK GUI edit

modified TTK GUI to work with the new events

Blounard committed:
Update startslide_utils.py

fixed startslides path

Blounard committed:
2 bugfixes

Blounard committed:
Update external_utils.py

Avoid clearing the shared memory before writing to it.
It could cause a rare issue where the external script would try to read the shared memory exactly in between the delete call, and the write call.

Also renaming variable to avoid confusion (bytes is a class in python)

Blounard committed:
TTK GUI bugfix + new buttons

TTK Save GUI bugfix
TTK GUI bugfix
TTK GUI new buttons
TTK GUI script end when ext window closes

Blounard committed:
Added module for Yaz1 management

Ghost can now be saved as Yaz1 compressed

Blounard committed:
Update _draw_info_display.py

Correctly clear History on stateload

Blounard committed:
bugfix compressed ghost

Blounard committed:
Create update_scripts.py

Script to update your scripts. Requires the requests module

Blounard committed:
update python stub

Epik95mkw committed:
Various fixes and updates

Epik95mkw committed:
Merge remote-tracking branch 'upstream/main'

Epik95mkw committed:
Restore input display changes

Blounard committed:
Merge pull request #3 from Epik95mkw/main

Various small fixes and updates

Blounard committed:
auto input optimizer v1

added a script that change inputs and keep improvement automatically.

Blounard committed:
bugfix infodisplay

Blounard committed:
Delete ttk_lib -backup.py

delete useless file

Blounard committed:
Update _TTK_Backup.py

Try to investigate on a bug.
This should be reverted when the bug will be fixed.
Trying to know why this https://blounard.s-ul.eu/1E62ITQK.png happens on loadstate seemingly randomly, and very very rarely

Blounard committed:
added requests to python packages installer

Blounard committed:
crash fix attempt

Attempt to fix a crash with the savestate event when the game is running, and memory is read

Blounard committed:
more infodisplay options

added player-ghost differences for rotation and speed

Blounard committed:
Update infodisplay_utils.py

added ahead/behind/tied text to infodisplay

Blounard committed:
infodisplay rkg buffer + erase rkg buffer

erase player buffer rkg : https://discord.com/channels/214574342137315328/735589843513638940/1362431341215350785

Blounard committed:
Update infodisplay_utils.py

forgot to add this file in the previous commit

Blounard committed:
AGC setting update

AGC setting now can accept operator in the delay (you can write '3+5' like in luacore)

Blounard committed:
agc fix aniamtion

inputs were playback 1f too soon

Blounard committed:
228 memory check event

Blounard committed:
agc save at exit crashfix

Blounard committed:
bug fix in settings

also added code for something later

Blounard committed:
python stubs update 230

Blounard committed:
Exact finish time

added exact finish time to infodisplay
also added a .txt file in Modules to keep track of EVA usage.

Blounard committed:
AGC load player

Blounard committed:
Improved freefly

Blounard committed:
auto input opti

some modification, and more options

Blounard committed:
small bugfix

Blounard committed:
Delete special.py

Blounard committed:
ttk gui update

added tooltip for buttons
added activate ghost hard
made a system for the ttk gui window to remember it's geometry when closing

Blounard committed:
Update Auto_Input_Optimizer.py

small bugfix

Epik95mkw committed:
Flag for no UI delay

Epik95mkw committed:
Upgraded superslide macro

Blounard committed:
Updated startslide

wiggle slide update
right slide are now loading left slides but perform mirror inputs

Epik95mkw committed:
Merge remote-tracking branch 'upstream/main'

Epik95mkw committed:
Boost panel timer

Blounard committed:
framedump extended

big script https://docs.google.com/document/d/e/2PACX-1vTXoEveB_1MZ3WizOpEWvZ-oyJMgg-3pRLGiNu-5vo853BMcrr8RY69REcTsheurI9qS2kfqrx1BZkT/pub

Blounard committed:
with the packages it's better

Blounard committed:
update wiggle slide, and some small fix

Blounard committed:
framedump script update

added mkw font to framedump script, and a bit more infodisplay option

Blounard committed:
bugfix

bugfix for the time needed to lose 1ms

Blounard committed:
debug script

not available for all regions

Epik95mkw committed:
QOL stuff

Epik95mkw committed:
Update macros after new EV discoveries

Epik95mkw committed:
Merge remote-tracking branch 'upstream/main'

Blounard committed:
Merge pull request #4 from Epik95mkw/main

Update EV macros

Blounard committed:
attempt to fix linux import

linux had broken import, this is an attempt at fixing it

Blounard committed:
Reverse previous commit

It actually breaks, i don't know why

Epik95mkw committed:
Attempt to fix import syntax for Linux

Blounard committed:
bugfix

bugfix in the framedump encoder

Blounard committed:
updater update

made the updater better. A single use of "Update_package_+_Scripts" does everything

Blounard committed:
tiny bugfix

vs code didn't like it for older python version, it was unused anyway

Blounard committed:
math geometry function

added math method on vec3 and quatf

Blounard committed:
geometry function

util function for getting some useful geometry. unused by main scripts currently, but could be useful

Blounard committed:
framedump osd extradisplay

added extra display for the osd framedump encoder.
it's a debug feature, you'd need to make the .rawtxt files yourself to use it

Epik95mkw committed:
Merge remote-tracking branch 'upstream/main'

Blounard committed:
Merge pull request #5 from Epik95mkw/main

Fix scripts for linux

Blounard committed:
mbp library

xAntares17 committed:
added input display images

xAntares17 committed:
implements width and color options, config_util changed accordingly

xAntares17 committed:
changed spelling: frame to frames

xAntares17 committed:
formatting

xAntares17 committed:
default color fixed

Blounard committed:
Merge pull request #6 from xAntares17/width-color-options

width & color options

xAntares17 committed:
does the recoloring once in main for better performance

xAntares17 committed:
restructuring

xAntares17 committed:
necessary name changes

Blounard committed:
Merge pull request #7 from xAntares17/width-color-options

Width color options

Blounard committed:
AGC copy input 1f delay fix

oops

Blounard committed:
Merge branch 'main' of https://github.com/Blounard/mkw-scripts

Blounard committed:
added comments

Blounard committed:
OSD update for the present event

adapting scripts to work with pycore 245

Blounard committed:
input display fix when not in race

xAntares17 committed:
creates additional fonts in the sizes specified in custom_text_scaling

xAntares17 committed:
implements custom text function, author display encoding changed to utf-8

xAntares17 committed:
added slash

Blounard committed:
Merge pull request #8 from xAntares17/custom-text

custom text

xAntares17 committed:
performance improvements

xAntares17 committed:
further optimization by selecting the analog stick parts depending on the bg

Blounard committed:
Merge pull request #9 from xAntares17/custom-text

performance improvements

xAntares17 committed:
made dpad edges less rough, added tutorial folder

xAntares17 committed:
fly in animation

xAntares17 committed:
reanchoring stick text depends on scale

xAntares17 committed:
Update input_display_util.py

kierio04 committed:
Add inactive outline colour to config

Also, renamed "unactive" labels to "inactive"

Blounard committed:
Merge pull request #11 from kierio04/main

Add inactive outline colour to config

Blounard committed:
Merge pull request #10 from xAntares17/custom-text

fly in + other fixes

xAntares17 committed:
pretty speedometer option added as gecko code doesnt use entry anim. fixed number style slash

xAntares17 committed:
adds pretty speedometer scaled font

Blounard committed:
Merge pull request #12 from xAntares17/custom-text

pretty speedometer

Blounard committed:
improved framdump script

framedump script now correctly find the filename for the framedump and audiodump
also added a script to delete framedumps

Blounard committed:
Update mbp_utils.py

Blounard committed:
Update _draw_info_display.py

Blounard committed:
Update framesequence.py

print an error to the osd + log when there is an invalid csv file opened

Blounard committed:
framedump script update

separate fade animation and fly animation

Blounard committed:
Update config_util.py

made comment case sensitive

Blounard committed:
Update config_util.py

added extra display to default config

Blounard committed:
Update encoder.py

allow for several infodisplay sections

xAntares17 committed:
fix fly out anim + other small changes

xAntares17 committed:
number quality improvement; letters mostly unchanged except saturation to 0

Blounard committed:
Merge pull request #13 from xAntares17/fixes-overhauls

fixes / font improvements

Blounard committed:
Update input_display_util.py

Blounard committed:
framedump script custom timings

you can now chose whenever a section appears and disappears

Blounard committed:
Update TTK_GUI.py

small bugfix
